File: Lv. 2/전력망을 둘로 나누기.py
def solution(n, wires):
    answer = []
    
    for i in range(len(wires)):
        new_wires = wires[:i] + wires[i+1:]
        wire_set = set(new_wires[0])
        check = [False for i in range(len(new_wires))]
        check[0] = True
        while True:
            stop = True
            for w in range(len(new_wires)):
                if check[w] == False:
                    if new_wires[w][0] in wire_set or new_wires[w][1] in wire_set:    
                        check[w] = True            
                        wire_set.add(new_wires[w][0])
                        wire_set.add(new_wires[w][1])
                        stop = False
            if stop:
                break
        answer.append(abs(2 * len(wire_set) - n))
    
    return min(answer)

File: Programmers/Lv. 2/전력망을 둘로 나누기.cpp
#include <string>
#include <vector>
#include <set>
#include <algorithm>

using namespace std;

bool IsIncluded(set<int> s, int i)
{
    auto iter = find(s.begin(), s.end(), i);
    return iter != s.end();
}

int solution(int n, vector<vector<int>> wires) {
    
    int m = 100;
    for (int w = 0; w < wires.size(); w++)
    {
        vector<vector<int>> new_wires(wires);
        new_wires.erase(new_wires.begin() + w);
        
        set<int> s1;
        s1.insert(new_wires[0][0]);
        s1.insert(new_wires[0][1]);
                
        bool checked[100] = {false, };
        
        int i = 1;
        while (i < new_wires.size())
        {                        
            if (!checked[i] && 
                (IsIncluded(s1, new_wires[i][0]) || IsIncluded(s1, new_wires[i][1])))
            {
                checked[i] = true;
                
                s1.insert(new_wires[i][0]);
                s1.insert(new_wires[i][1]);    
                
                i = 1;    
                continue;
            }
            i++;
        }       
        
        int a = max({s1.size(), n - s1.size()});
        int b = min({s1.size(), n - s1.size()});
        
        m = min({ m,  a - b });
    }    
    
    return m;
}